### PR TITLE
Make JSONField compatible with python types

### PR DESCRIPTION
**Pull request by cwisecarver**
@sloria I wouldn't say this is the cleanest implementation in the land but for the types specified it works. The last commit is the one that actually fixes things. I'm happy to have suggestions on how to do it better. Several hours of googling didn't serve me any easy solutions.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -23,6 +23,7 @@
         'bleach>=1.4.1',
         'pytz>=2014.9',
         'django-dirtyfields>=1.1',
+        'ujson>=1.35',
```
Comment (sloria) [inline setup.py]: This dependency can be removed correct, since you're no longer using ujson?

File: osf_models/utils/datetime_aware_jsonfield.py [hunk 2]
```diff
@@ -1,31 +1,104 @@
...
+            return dict(type='encoded_time', value=o.isoformat())
+        elif isinstance(o, Decimal):
+            return dict(type='encoded_decimal', value=str(o))
+        return super(DateTimeAwareJSONEncoder, self).default(o)
+
+
+def decode_datetime_objects(nested_value):
+
+    if isinstance(nested_value, list):
+        new_list = list()
```
Comment (sloria) [inline osf_models/utils/datetime_aware_jsonfield.py]: Minor: can be refactored to 

```python
return [decode_datetime_objects(item) for item in nested_value]
```

File: osf_models/utils/datetime_aware_jsonfield.py [hunk 3]
```diff
@@ -1,31 +1,104 @@
...
+        for item in nested_value:
+            new_list.append(decode_datetime_objects(item))
+        return new_list
+    elif isinstance(nested_value, dict):
+        new_value = nested_value
```
Comment (sloria) [inline osf_models/utils/datetime_aware_jsonfield.py]: This will cause this function to modify the input in-place; is that intentional?

**Comment by sloria** [comment]
Just a few minor things. +1 to fixing DateTimeAwareJSONField. If this is how we gotta do it, so be it.

File: osf_models/utils/datetime_aware_jsonfield.py (hunk 2 above)
Comment (cwisecarver) [inline osf_models/utils/datetime_aware_jsonfield.py]: Done.

File: osf_models/utils/datetime_aware_jsonfield.py (hunk 3 above)
Comment (cwisecarver) [inline osf_models/utils/datetime_aware_jsonfield.py]: I thought I was avoiding a "changed dictionary while iterating" error. I just removed new_value entirely and instead just modify nested_value. Should be more obvious that it's intentional now.

File: osf_models/tests.py [hunk 4]
```diff
@@ -1,3 +1,68 @@
+import datetime as dt
+import json
+from decimal import Decimal
+
 from django.test import TestCase
+from osf_models.utils.datetime_aware_jsonfield import DateTimeAwareJSONEncoder, decode_datetime_objects
+
+
+class DateTimeAwareJSONFieldTests(TestCase):
```
Comment (cwisecarver) [inline osf_models/tests.py]: I'll move them in a separate PR. I'm in a funny state with my different repos right now.

**Pull request merged**